### PR TITLE
Fix more jsdoc/tag-lines ESLint errors

### PR DESCRIPTION
**Pull request by jsnajdr**
Follow up to #84137 which fixes many more `jsdoc/tag-lines` and handful of `jsdoc/no-defaults` errors. #84137 fixed only `.jsx` files, but we have also `.js`, `.ts` and `.tsx`!

Almost all the changes are removing a line in a jsdoc comment.

The second commit fixes a few more little errors. Two violations of `jsdoc/require-returns-check` and one disabling of `@typescript-eslint/no-this-alias`.

### CODE REVIEW

File: client/jetpack-cloud/controller.ts [hunk 1]
```diff
@@ -165,15 +164,11 @@ function recordNoVisibleJetpackSitesPageView(
 
 /**
  * Show dedicated screen if user has no Jetpack site, or no visible Jetpack site
- *
- * @param {string} siteFragment Parsed site slug
+ * @param {string|undefined} siteFragment Parsed site slug
```
Comment (tyxla) [inline client/jetpack-cloud/controller.ts:167]: I thought we don't need JSDoc when we have types. Maybe we can remove the types from these `@param` tags.

File: client/jetpack-cloud/controller.ts [hunk 2]
```diff
@@ -188,11 +183,12 @@ export function noSite(
 		recordNoVisibleJetpackSitesPageView( context, siteFragment );
 		return true;
 	}
+
+	return false;
```
Comment (tyxla) [inline client/jetpack-cloud/controller.ts:187]: 👍

File: client/my-sites/plans/jetpack-plans/build-card-features-from-item.ts [hunk 3]
```diff
@@ -10,9 +10,8 @@ import type { Plan, Feature } from '@automattic/calypso-products';
 
 /**
  * Builds the feature item of a product card, from a feature key.
- *
- * @param {Feature[]|Feature} featureKey Key of the feature
- * @returns {SelectorProductFeaturesItem} Feature item
+ * @param {Feature} featureKey Key of the feature
```
Comment (tyxla) [inline client/my-sites/plans/jetpack-plans/build-card-features-from-item.ts:13]: Same as above - for typed arguments we should be able to remove the types from JSDoc.

**Review by tyxla — APPROVED** [review]
Those are many files to scan, but luckily the changes are mundane and easy to quickly glance through.

LGTM! I've left a few comments, but nothing blocking.

🚀

File: client/jetpack-cloud/controller.ts (hunk 1 above)
Comment (jsnajdr) [reply client/jetpack-cloud/controller.ts:167]: I agree with you, but this is not about one or two functions. It's a large scale issue in Calypso TypeScript code that types are duplicate in TS and in JSDoc. Better solve it separately about some wider discussion about how to best combine TS and JSDoc.

Comment (tyxla) [reply client/jetpack-cloud/controller.ts:167]: Of course, just wanted to point it out while we're here 👍

**Pull request merged**